### PR TITLE
WEB-8081 the-wall seat functionality

### PR DESCRIPTION
**Pull request by TomFirth**
#### What are the links to relevant tickets?
https://hxshortbreaks.atlassian.net/browse/WEB-8079
https://hxshortbreaks.atlassian.net/browse/WEB-8080
https://hxshortbreaks.atlassian.net/browse/WEB-8081
#### What does this PR do?
Using the Transformer config, this adds Gold, Silver or Bronze to the relevant packageRate depending on their existing priceBand.
#### What functional/unit tests does this PR have?
none
#### Screenshots (if appropriate)
---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [ ] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [x] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.

#### Reviewer 1 @t1gr0u
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

#### Reviewer 2 @kevinhodges
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

File: helpers/filterSeats.js [hunk 1]
```diff
@@ -58,12 +59,32 @@ module.exports = function(dust) {
       });
     }
 
+    // use Transformer show config to add Gold, Silver or Bronze to packageRate
+    function assignColoursToBands(packageRate) {
+      // assign gold, silver, bronze to packages depending on their current priceBand
+      _.forEach(transformer, function(sectionValue, sectionKey) {
```
Comment (t1gr0u) [inline helpers/filterSeats.js]: you may want to check that Transformer is available before

Comment (t1gr0u) [inline helpers/filterSeats.js]: `transformer` is maybe not the correct variable name, maybe `bandColours`, as you are only using the result from Transformer

**Comment by kevinhodges** [comment]
Code looks good, I haven't been able to test it i'm afraid though

File: helpers/filterSeats.js [hunk 2]
```diff
@@ -58,12 +59,35 @@ module.exports = function(dust) {
...
+    }
+
     function loopReplySections(packageRate) {
       // For every section in the stubbed out reply
       // check if the current package rate is in the current section
       _.forEach(reply, function(item, i) {
         // If so then we want to add this packageRate under the theatre section.
         if (ticketRates[packageRate.links.ticketRates.ids].section === item.name) {
+          // WEB-8081
+          if (bandColours !== null) {
```
Comment (kevinhodges) [inline helpers/filterSeats.js]: I would change this to check any kind of `falsy`, in my test `bandColours` is `undefined` and we still execute the `assignColoursRoBands` function. Maybe something like 

```
if (bandColours) {
```

File: helpers/filterSeats.js [hunk 3]
```diff
@@ -58,12 +59,35 @@ module.exports = function(dust) {
       });
     }
 
+    // use Transformer show config to add Gold, Silver or Bronze to packageRate
+    function assignColoursToBands(packageRate) {
+      // assign gold, silver, bronze to packages depending on their current priceBand
+      _.forEach(bandColours, function(sectionValue, sectionKey) {
+        // check for match between transformer and existing data
+        // check for section that might not have full title ( e.g. Grand Circle in Grand Circle (Left) )
+        if (ticketRates[packageRate.ids].section === sectionKey || ticketRates[packageRate.ids].section.indexOf( sectionKey ) > -1 ) {
```
Comment (kevinhodges) [inline helpers/filterSeats.js]: I would challenge consuming this from global scope, this makes things tricky to test, passing ticketRates in as a 2nd param makes things a lot more readable too

File: helpers/filterSeats.js [hunk 4]
```diff
@@ -58,12 +59,35 @@ module.exports = function(dust) {
...
+          // match transformer config to existing seat section
+          _.forEach(sectionValue, function(bandValue, bandKey) {
+            if (ticketRates[packageRate.ids].priceBand === bandKey) {
+              // match transformer config to existing priceBand
+              ticketRates[packageRate.ids].colour = bandValue;
```
Comment (kevinhodges) [inline helpers/filterSeats.js]: can we bin one of these assigns please, don't want the same info twice on the api reply

File: helpers/filterSeats.js [hunk 5]
```diff
@@ -58,12 +59,35 @@ module.exports = function(dust) {
       });
     }
 
+    // use Transformer show config to add Gold, Silver or Bronze to packageRate
+    function assignColoursToBands(packageRate) {
```
Comment (kevinhodges) [inline helpers/filterSeats.js]: this isn't a `packageRate`, it's a `ticketRate`

**Pull request merged**